### PR TITLE
Bean validation integration

### LINKED ISSUE #482: MicroProfile Open API working with Bean Validation
As part of the program plan 2021, the community would like to see some integration between Open API and Bean Validation. Please comment on this issue on what areas can Bean Validation help with MP Open API. What about schema validation?

### PR DESCRIPTION
**Pull request by Azquelt**
- Schema enhanced by certain bean validation annotations
- Can be overridden by user
- Can be disabled by config

Fixes #482

### CODE REVIEW

**Comment by Azquelt** [comment]
There's a commit to autoformat the code before my changes, since maven does this on every build. I  forgot to run a full build before my last PR

**Comment by Azquelt** [comment]
I haven't quite got all of these passing on the smallrye impl yet but will work on that tomorrow.

File: spec/src/main/asciidoc/microprofile-openapi-spec.asciidoc [hunk 1]
```diff
@@ -573,6 +577,48 @@ post:
 
 For more samples please see the https://github.com/eclipse/microprofile-open-api/wiki[MicroProfile Wiki].
 
+==== Jakarta Bean Validation Annotations
+
+In some cases, additional schema restrictions can be inferred from Jakarta Bean Validation annotations and used to enhance the generated OpenAPI document.
+
+Vendors are required to process Jakarta Bean Validation annotations and add the properties listed in the table below to the schema model when:
```
Comment (Emily-Jiang) [inline spec/src/main/asciidoc/microprofile-openapi-spec.asciidoc:584]:
```suggestion
Implementors are required to process Jakarta Bean Validation annotations and add the properties listed in the table below to the schema model when:
```

**Review by Emily-Jiang — APPROVED** [review]

File: spec/src/main/asciidoc/microprofile-openapi-spec.asciidoc (hunk 1 above)
Comment (Azquelt) [reply spec/src/main/asciidoc/microprofile-openapi-spec.asciidoc:584]: I'm happy to change this, but "vendors" matches the language used in the rest of the spec, so I'll also raise an issue to consistently talk about implementers (or maybe implementations would be better)

File: tck/src/main/java/org/eclipse/microprofile/openapi/apps/beanvalidation/BeanValidationData.java [hunk 2]
```diff
@@ -0,0 +1,259 @@
...
+    private int negativeInt;
+
+    @NegativeOrZero
+    private int negativeOrZeroInt;
+
+    @Positive
+    private int positiveInt;
+
+    @PositiveOrZero
+    private int positiveOrZeroInt;
```
Comment (MikeEdgar) [inline tck/src/main/java/org/eclipse/microprofile/openapi/apps/beanvalidation/BeanValidationData.java:88]: The intent with using the "exclusive" booleans with positive/negative wasn't immediately obvious to me due to the use of integers. Good catch.

**Review by MikeEdgar — APPROVED** [review]

**Pull request merged**